### PR TITLE
[improverment]:The install.sh script does not check whether the user has sudo privileges

### PR DESCRIPTION
**Pull request by ZYJBigData**
## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

### CODE REVIEW

File: script/install.sh [hunk 1]
```diff
@@ -22,14 +22,21 @@ workDir=`cd ${workDir};pwd`
...
 if [ ! -d $installPath ];then
-  sudo mkdir -p $installPath
-  sudo chown -R $deployUser:$deployUser $installPath
+   echo "sudo enabled = ${sudoEnable}"
+   if [ ${sudoEnable} = true ]; then
+      sudo mkdir -p $installPath
+      sudo chown -R $deployUser:$deployUser $installPath
+    else
+        mkdir -p $installPath
+        chown -R $deployUser:$deployUser $installPath
```
Comment (SbloodyS) [inline script/install.sh]: Please use the same format.

File: script/env/install_env.sh [hunk 2]
```diff
@@ -59,3 +59,6 @@ deployUser=${deployUser:-"dolphinscheduler"}
 
 # The root of zookeeper, for now DolphinScheduler default registry server is zookeeper.
 zkRoot=${zkRoot:-"/dolphinscheduler"}
+
+#whether the installation user has sudo permission
+sudoEnable=${sudoEnable:-"true"}
```
Comment (SbloodyS) [inline script/env/install_env.sh:64]: It's better to add this to the document.

**Comment by EricGao888** [comment]
@ZYJBigData Hi, could you please rebase the code and push it again? It helps restart the CI. BTW, it would be better if you could keep the PR title complete : ) Thx~

File: script/install.sh (hunk 1 above)
Comment (chengshiwen) [reply script/install.sh]: code styles +1

File: script/env/install_env.sh (hunk 2 above)
Comment (chengshiwen) [reply script/env/install_env.sh:64]: Considering the code style, It would be better to use `# whether` instead of `#whether`

**Comment by zhuxt2015** [comment]
I have a little doubt about that If the user does not have sudo permissions, should the worker be launched, because the task cannot be started without sudo permissions

**Comment by EricGao888** [comment]
> I have a little doubt about that If the user does not have sudo permissions, should the worker be launched, because the task cannot be started without sudo permissions

+1, instead of continuing, maybe the script should exit and output a message to instruct users to set up sudo permissions with a link to related doc or something. WDYT @ZYJBigData @chengshiwen @SbloodyS

**Comment by SbloodyS** [comment]
> > I have a little doubt about that If the user does not have sudo permissions, should the worker be launched, because the task cannot be started without sudo permissions
> 
> +1, instead of continuing, maybe the script should exit and output a message to instruct users to set up sudo permissions with a link to related doc or something. WDYT @ZYJBigData @chengshiwen @SbloodyS

It seems that we have supported `non sudo`.
https://github.com/apache/dolphinscheduler/blob/5b429d29b88de4064cb2e87082047f8337da94a2/dolphinscheduler-common/src/main/resources/common.properties#L86-L87

File: script/env/install_env.sh [hunk 3]
```diff
@@ -59,3 +59,6 @@ deployUser=${deployUser:-"dolphinscheduler"}
 
 # The root of zookeeper, for now DolphinScheduler default registry server is zookeeper.
 zkRoot=${zkRoot:-"/dolphinscheduler"}
+
+# whether the installation user has sudo permission
+sudoEnable=${sudoEnable:-"true"}
```
Comment (EricGao888) [inline script/env/install_env.sh:64]: Could you please update the docs about this? Thx

**Comment by EricGao888** [comment]
> > > I have a little doubt about that If the user does not have sudo permissions, should the worker be launched, because the task cannot be started without sudo permissions
> > 
> > 
> > +1, instead of continuing, maybe the script should exit and output a message to instruct users to set up sudo permissions with a link to related doc or something. WDYT @ZYJBigData @chengshiwen @SbloodyS
> 
> It seems that we have supported `non sudo`.
> 
> https://github.com/apache/dolphinscheduler/blob/5b429d29b88de4064cb2e87082047f8337da94a2/dolphinscheduler-common/src/main/resources/common.properties#L86-L87

In that case, LGTM as long as related docs are updated.

File: script/env/install_env.sh (hunk 3 above)
Comment (ZYJBigData) [reply script/env/install_env.sh:64]: @EricGao888  OK soon, Thanks

File: script/install.sh [hunk 4]
```diff
@@ -22,14 +22,21 @@ workDir=`cd ${workDir};pwd`
...
-  sudo mkdir -p $installPath
-  sudo chown -R $deployUser:$deployUser $installPath
+  echo "sudo enabled = ${sudoEnable}"
+   if [ ${sudoEnable} = true ]; then
+      sudo mkdir -p $installPath
+      sudo chown -R $deployUser:$deployUser $installPath
+    else
+      mkdir -p $installPath
+      chown -R $deployUser:$deployUser $installPath
+    fi
```
Comment (zhongjiajie) [inline script/install.sh:39]: Why we need this?  should we directly remove the `sudo` and require user have permission of `installPath`?

**Pull request closed without merging**